src/exit_codes.rs

```diff
@@ -37,6 +43,22 @@ mod tests {
         assert_eq!(merge_exitcodes(&[]), ExitCode::Success);
     }
 
+    #[test]
+    fn success_when_has_match() {
+        assert_eq!(
+            merge_exitcodes(&[ExitCode::HasMatch(true)]),
+            ExitCode::Success
+        );
+    }
+
+    #[test]
+    fn general_error_if_has_no_match() {
+        assert_eq!(
+            merge_exitcodes(&[ExitCode::HasMatch(false)]),
+            ExitCode::GeneralError
+        );
+    }
```

Comment:
Thank you for adding tests, but `merge_exitcodes` is really only relevant for job execution. `HasMatch(_)` is not an output that could result from executing a job.

I think we can remove these tests, what do you think?

Maybe this is a sign that we are merging two functionalities here which are attached to a single datatype (`ExitCode`). Not sure if there is a cleaner strategy for implementing this though.

Reply:
Oh, I see. I think you're right; not only is `merge_exitcodes` not relevant here, but the integration test I added already covers this functionality. I'll go ahead and remove these two unit tests for now.